File: 30-days-of-code/java/Day20.java
import java.io.*;
import java.util.*;
import java.text.*;
import java.math.*;
import java.util.regex.*;

public class Day20 {

   public static void main(String[] args) {
      Scanner in = new Scanner(System.in);
      int n = in.nextInt();
      int a[] = new int[n];
      for (int a_i = 0; a_i < n; a_i++) {
         a[a_i] = in.nextInt();
      }
      int numberOfSwaps = 0;
      for (int i = 0; i < n; i++) {
         for (int j = 0; j < n - 1; j++) {

            if (a[j] > a[j + 1]) {
               int tmp = a[j];
               a[j] = a[j + 1];
               a[j + 1] = tmp;
               numberOfSwaps++;
            }
         }

         if (numberOfSwaps == 0) {
            break;
         }
      }
      System.out.printf("Array is sorted in %d swaps.\n", numberOfSwaps);
      System.out.printf("First Element: %d\n", a[0]);
      System.out.printf("Last Element: %d\n", a[a.length - 1]);
   }
}

File: 30-days-of-code/go/Day20.go
package main

import "fmt"

func main() {
	var n int
	fmt.Scanf("%d", &n)
	arr := make([]int, n)
	//read input
	for i := 0; i < n; i++ {
		var val int
		fmt.Scanf("%d", &val)
		arr[i] = val
	}
	numberOfSwaps := 0
	for i := 0; i < n; i++ {
		for j := 0; j < n-1; j++ {
			if arr[j] > arr[j+1] {
				//swap
				arr[j+1], arr[j] = arr[j], arr[j+1]
				numberOfSwaps++
			}
		}
		if numberOfSwaps == 0 {
			break
		}
	}
	fmt.Printf("Array is sorted in %d swaps.\n", numberOfSwaps)
	fmt.Printf("First Element: %d\n", arr[0])
	fmt.Printf("Last Element: %d\n", arr[len(arr)-1])
}
